### PR TITLE
drivers/periph_rtt: add periph_rtt_set_counter  feautre

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

`examples/gnrc_networking_mac` currently uses a whitelist to manage supported boards. It seems that missing `periph_rtt` features are the reason that it doesn't work on some boards. The functions in questions seem to be `rtt_get_alarm()` and `rtt_set_counter()`. The former is not implemented for STM32 except STM32F1 for no reason other than the test application not testing it. The latter cannot be easily implemented for STM32 except STM32F1.

This should solve the issuses by:

1. Providing `rtt_get_alarm()` for STM32
2. Also testing `rtt_get_alarm()` in `tests/periph_rtt` to not let others get away with not providing it
3. Adding `periph_rtt_set_counter` as feature. As follow up `examples/gnrc_networking_mac` should get rid of the whitelist and instead be able to just depend on that feature
4. Extending the test to also cover `rtt_set_counter()` if the optional feature `periph_rtt_set_counter` is provided

### Testing procedure

`tests/periph_rtt` now got you covered :-)

### Issues/PRs references

Noticed while testing https://github.com/RIOT-OS/RIOT/pull/16100

### CODE REVIEW

File: cpu/atmega_common/Makefile.features [hunk 1]
```diff
@@ -3,6 +3,10 @@ include $(RIOTCPU)/avr8_common/Makefile.features
 # common feature are defined in avr8_common/Makefile.features
 # Only add Additional features
 
+ifneq (,$(filter periph_rtt,$(FEATURES_PROVIDED)))
+  FEATURES_PROVIDED += periph_rtt_set_counter
+endif
```
Comment (benpicco) [inline cpu/atmega_common/Makefile.features:8]: We can set this pseudofeature unconditionally, see #15763

Comment (maribu) [reply cpu/atmega_common/Makefile.features:8]: OK.

File: tests/periph_rtt/main.c [hunk 2]
```diff
@@ -75,10 +75,32 @@ int main(void)
...
     last = (now + TICKS_TO_WAIT) & RTT_MAX_VALUE;
     printf("Setting initial alarm to now + 5 s (%" PRIu32 ")\n", last);
     rtt_set_alarm(last, cb, 0);
 
+    if (rtt_get_alarm() != last) {
+        puts("Error: rtt_get_alarm() not working");
+        return 1;
+    }
+        else {
+            puts("rtt_get_alarm() PASSED");
```
Comment (benpicco) [inline tests/periph_rtt/main.c:101]: Indentation mishap

Comment (maribu) [reply tests/periph_rtt/main.c:101]: Fixed

**Comment by benpicco** [comment]
Please squash

**Review by benpicco — APPROVED** [review]

**Comment by maribu** [comment]
I always forget KConfig :-(

**Comment by benpicco** [comment]
Please squash again

File: tests/periph_rtt/Makefile [hunk 3]
```diff
@@ -2,6 +2,7 @@ BOARD ?= samr21-xpro
 include ../Makefile.tests_common
 
 FEATURES_REQUIRED = periph_rtt
+FEATURES_OPTIONAL += periph_rtt_set_counter
```
Comment (leandrolanzieri) [inline tests/periph_rtt/Makefile:5]: I think here we need a Kconfig to `imply` this module when it's available. Something like

https://github.com/RIOT-OS/RIOT/blob/dfb2c21570eb31bd9003eff20e1a7d2d1365c7d4/tests/periph_flashpage/Kconfig#L8-L13

Comment (maribu) [reply tests/periph_rtt/Makefile:5]: Thanks. This did the trick (at least locally it seems to work now). I directly amended the change into the last commit.

**Comment by maribu** [comment]
Finally: All green :-)

**Comment by maribu** [comment]
Thanks :-)

**Comment by fjmolinas** [comment]
Was the testing procedure ran? Why do we have one of the test is not run? Murdock is not even running the test the `BOARD`s, and so know the test fails on `nrf52`, `cc2538`, its probably exposing bugs by increasing the coverage, but still

- nrf52
```
main(): This is RIOT! (Version: riot/pr/16137)

RIOT RTT low-level driver test
RTT configuration:
RTT_MAX_VALUE: 0x00ffffff
RTT_FREQUENCY: 1024

Testing the tick conversion
Trying to convert 1 to seconds and back
Trying to convert 256 to seconds and back
Trying to convert 65536 to seconds and back
Trying to convert 16777216 to seconds and back
Trying to convert 2147483648 to seconds and back
All ok

Initializing the RTT driver
This test will now display 'Hello' every 5 seconds

RTT now: 4
Setting RTT timer to 1337
RTT now: 7
ERROR: rtt_set_counter() failed (off by more than 1 ms)
Timeout in expect script at "child.expect(r'Setting initial alarm to now \+ {} s \(\d+\)'" (tests/periph_rtt/tests/01-run.py:26)

make: *** [/builds/workspace/RIOT.git.jk/makefiles/tests/tests.inc.mk:22: test] Error 1

```

- cc2538

```
main(): This is RIOT! (Version: riot/pr/16137)

RIOT RTT low-level driver test
RTT configuration:
RTT_MAX_VALUE: 0xffffffff
RTT_FREQUENCY: 32768

Testing the tick conversion
Trying to convert 1 to seconds and back
Trying to convert 256 to seconds and back
Trying to convert 65536 to seconds and back
Trying to convert 16777216 to seconds and back
Trying to convert 2147483648 to seconds and back
All ok

Initializing the RTT driver
This test will now display 'Hello' every 5 seconds

RTT now: 3210119170
Setting RTT timer to 1337
RTT now: 4294965959
ERROR: rtt_set_counter() failed (off by more than 1 ms)
Timeout in expect script at "child.expect(r'Setting initial alarm to now \+ {} s \(\d+\)'" (tests/periph_rtt/tests/01-run.py:26)

make: *** [/builds/workspace/RIOT.git.jk/makefiles/tests/tests.inc.mk:22: test] Error 1
```

**Comment by benpicco** [comment]
Oh sorry, this seemed like a simple change and I only tested on sam0.
Should have checked better, e.g. `nrf5x` has

```C
void rtt_set_counter(uint32_t counter)
{
    (void) counter;
    /* not supported by the nRF5x */
}
```

I guess we can't assume API implementations that have no tests to work across all implementations.

**Comment by maribu** [comment]
The fix ia straight forward: Drop the feature for the nRF. Give me a minute to open the PR.

**Comment by maribu** [comment]
> Oh sorry, this seemed like a simple change and I only tested on sam0.
> Should have checked better

I disagree that we need to test an extend unit test before merging it on every board. This PR is about improving the unit test, not about fixing bugs present in `master`.

This seemingly is a regression (a test previously passed now fails), but IMO it is an improvement. A false positive test result is now turned into a correct negative - IMO this is an improvement.

**Comment by fjmolinas** [comment]
> This seemingly is a regression (a test previously passed now fails), but IMO it is an improvement. A false positive test result is now turned into a correct negative - IMO this is an improvement.

I'm not arguing against this, this PR did of course not break anything but expose something, so I'm not arguing against introducing a test that might be broken on some `BOARD`s, if a test exposes an issue let's merge it. What I don't like is not being explicit on:

- what was tested
- how it was tested

I don't think it needs to be tested on every BOARD, but from looking at this PR I would have thought this was not tested at all:

- no Murdock `run-test`
- no test logs from a contributor or reviewer

And this IMO is not fine.

**Pull request merged**